### PR TITLE
fix(plasma-web, plasma-b2c, plasma-ui): fix single/multiple lines in Checkbox/Radiobox

### PR DESCRIPTION
**Pull request by kayman233**
Произошло два изменения:

1. поправили `label` и `description` у компонентов `Checkbox` и `Radiobox`.
    **Было**:
        Снимок экрана 2023-05-22 в 15 42 40
        - `description` был многострочный, а `label` однострочный, при этом `description` подстраивался под ширину `label`, из-за чего он мог неправильно отображаться.
        - Также не отображалось троеточие.

    **Стало**:
        Снимок экрана 2023-05-22 в 15 42 43
        - Теперь `label` и `description` отображаются единым образом.
        - Троеточие отображается корректно.

2. добавили свой-во `singleLine`, с помощью которого появилась возможность ограничить контент одной строкой:
    - Без свойства `singleLine` (по умолчанию): 
    Снимок экрана 2023-05-19 в 16 13 55

    - Со свойством `singleLine`: 
    Снимок экрана 2023-05-19 в 16 13 45

:baby_chick: Download canary assets:

[color_sberHealth_ios-swift--canary.511.5024620262.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.511.5024620262.swift)  
[color_sberHealth_kotlin--canary.511.5024620262.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.511.5024620262.kt)  
[color_sberHealth_react-native--canary.511.5024620262.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.511.5024620262.ts)  
[color_sbermarket_business_ios-swift--canary.511.5024620262.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_ios-swift--canary.511.5024620262.swift)  
[color_sbermarket_business_kotlin--canary.511.5024620262.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_kotlin--canary.511.5024620262.kt)  
[color_sbermarket_business_react-native--canary.511.5024620262.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_react-native--canary.511.5024620262.ts)  
[color_sbermarket_ios-swift--canary.511.5024620262.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.511.5024620262.swift)  
[color_sbermarket_kotlin--canary.511.5024620262.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.511.5024620262.kt)  
[color_sbermarket_metro_ios-swift--canary.511.5024620262.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_ios-swift--canary.511.5024620262.swift)  
[color_sbermarket_metro_kotlin--canary.511.5024620262.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_kotlin--canary.511.5024620262.kt)  
[color_sbermarket_metro_react-native--canary.511.5024620262.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_react-native--canary.511.5024620262.ts)  
[color_sbermarket_react-native--canary.511.5024620262.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.511.5024620262.ts)  
[color_sbermarket_selgros_ios-swift--canary.511.5024620262.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_ios-swift--canary.511.5024620262.swift)  
[color_sbermarket_selgros_kotlin--canary.511.5024620262.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_kotlin--canary.511.5024620262.kt)  
[color_sbermarket_selgros_react-native--canary.511.5024620262.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_react-native--canary.511.5024620262.ts)  
[color_sbermarket_wlbusiness_ios-swift--canary.511.5024620262.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_ios-swift--canary.511.5024620262.swift)  
[color_sbermarket_wlbusiness_kotlin--canary.511.5024620262.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_kotlin--canary.511.5024620262.kt)  
[color_sbermarket_wlbusiness_react-native--canary.511.5024620262.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_react-native--canary.511.5024620262.ts)  
[color_sberonline_ios-swift--canary.511.5024620262.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_ios-swift--canary.511.5024620262.swift)  
[color_sberonline_kotlin--canary.511.5024620262.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_kotlin--canary.511.5024620262.kt)  
[color_sberonline_react-native--canary.511.5024620262.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_react-native--canary.511.5024620262.ts)  
[color_sberprime_ios-swift--canary.511.5024620262.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.511.5024620262.swift)  
[color_sberprime_kotlin--canary.511.5024620262.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.511.5024620262.kt)  
[color_sberprime_react-native--canary.511.5024620262.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.511.5024620262.ts)  
[shadow_sbermarket_react-native--canary.511.5024620262.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.511.5024620262.ts)  
[typo_mage_ios-swift--canary.511.5024620262.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.511.5024620262.swift)  
[typo_mage_kotlin--canary.511.5024620262.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.511.5024620262.kt)  
[typo_mage_react-native--canary.511.5024620262.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.511.5024620262.ts)  
[typo_plasma_ios-swift--canary.511.5024620262.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.511.5024620262.swift)  
[typo_plasma_kotlin--canary.511.5024620262.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.511.5024620262.kt)  
[typo_plasma_react-native--canary.511.5024620262.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.511.5024620262.ts)  
[typo_ruler_ios-swift--canary.511.5024620262.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.511.5024620262.swift)  
[typo_ruler_kotlin--canary.511.5024620262.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.511.5024620262.kt)  
[typo_ruler_react-native--canary.511.5024620262.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.511.5024620262.ts)  
[typo_sage_ios-swift--canary.511.5024620262.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.511.5024620262.swift)  
[typo_sage_kotlin--canary.511.5024620262.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.511.5024620262.kt)  
[typo_sage_react-native--canary.511.5024620262.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.511.5024620262.ts)  
[typo_sbermarket_ios-swift--canary.511.5024620262.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.511.5024620262.swift)  
[typo_sbermarket_kotlin--canary.511.5024620262.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.511.5024620262.kt)  
[typo_sbermarket_react-native--canary.511.5024620262.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.511.5024620262.ts)  
[typo_soulmate_ios-swift--canary.511.5024620262.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.511.5024620262.swift)  
[typo_soulmate_kotlin--canary.511.5024620262.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.511.5024620262.kt)  
[typo_soulmate_react-native--canary.511.5024620262.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.511.5024620262.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.214.0-canary.511.5024620262.0
  npm install @salutejs/plasma-core@1.119.0-canary.511.5024620262.0
  npm install @salutejs/plasma-hope@1.214.0-canary.511.5024620262.0
  npm install @salutejs/plasma-icons@1.148.0-canary.511.5024620262.0
  npm install @salutejs/plasma-temple@1.163.0-canary.511.5024620262.0
  npm install @salutejs/plasma-ui@1.195.0-canary.511.5024620262.0
  npm install @salutejs/plasma-web@1.214.0-canary.511.5024620262.0
  npm install @salutejs/plasma-sb-utils@0.117.0-canary.511.5024620262.0
  # or 
  yarn add @salutejs/plasma-b2c@1.214.0-canary.511.5024620262.0
  yarn add @salutejs/plasma-core@1.119.0-canary.511.5024620262.0
  yarn add @salutejs/plasma-hope@1.214.0-canary.511.5024620262.0
  yarn add @salutejs/plasma-icons@1.148.0-canary.511.5024620262.0
  yarn add @salutejs/plasma-temple@1.163.0-canary.511.5024620262.0
  yarn add @salutejs/plasma-ui@1.195.0-canary.511.5024620262.0
  yarn add @salutejs/plasma-web@1.214.0-canary.511.5024620262.0
  yarn add @salutejs/plasma-sb-utils@0.117.0-canary.511.5024620262.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-511/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-511/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-511/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-511/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-511/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-511/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-511/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-511/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-511/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-511/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-511/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-511/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-511/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-511/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-511/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-511/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Review by Yakutoc — APPROVED** [review]

File: cypress/snapshots/ui/components/Checkbox/CheckboxBase.component-test.tsx/plasma-core Checkbox -- squeezes.snap.png
Comment (Yeti-or) [inline cypress/snapshots/ui/components/Checkbox/CheckboxBase.component-test.tsx/plasma-core Checkbox -- squeezes.snap.png:1]: я не очень понял, если мы ничего не ломаем, почему поведение изменилось?

File: packages/plasma-core/src/components/Basebox/Basebox.tsx [hunk 1]
```diff
@@ -58,9 +62,18 @@ export const StyledContent = styled.div`
     flex-direction: column;
     margin-left: 0.75rem;
 `;
-export const StyledLabel = styled(Body1)`
-    ${applyEllipsis}
+export const StyledLabel = styled(Body1)<{ singleLine?: boolean }>`
+    ${({ singleLine }) =>
+        singleLine &&
+        css`
```
Comment (Yeti-or) [inline packages/plasma-core/src/components/Basebox/Basebox.tsx:68]: тут не надо наш миксин исползовать?

https://github.com/salute-developers/plasma/blob/master/packages/plasma-core/src/mixins/applyEllipsis.ts#L3

File: packages/plasma-core/src/components/Checkbox/Checkbox.component-test.tsx [hunk 2]
```diff
@@ -108,6 +108,7 @@ describe('plasma-core: Checkbox', () => {
                         disabled
                         label="Checkbox with a very very very very very long label"
                         description="Checkbox description with a very very very very very long text"
+                        singleLine
```
Comment (Yeti-or) [inline packages/plasma-core/src/components/Checkbox/Checkbox.component-test.tsx:111]: не давай отдельный кейс на это сделаем

File: packages/plasma-hope/src/components/Checkbox/Checkbox.tsx [hunk 3]
```diff
@@ -141,6 +141,7 @@ export const StyledTrigger = styled(BaseboxTrigger)<{ outlineRadius?: string } &
     `}
 `;
 export const StyledContent = styled(BaseboxContent)<SizeProps>`
+    overflow: hidden;
```
Comment (Yeti-or) [inline packages/plasma-hope/src/components/Checkbox/Checkbox.tsx:144]: ну вот вопрос опять про наш миксин

**Review by Yeti-or — COMMENTED** [review]
А почему 
```
Поправлено поведение label и description у компонентов Checkbox и Radiobox ...
```

разве мы просто не добавляем новое поведени?
что именно мы правим

File: packages/plasma-core/src/components/Basebox/Basebox.tsx (hunk 1 above)
Comment (kayman233) [reply packages/plasma-core/src/components/Basebox/Basebox.tsx:68]: не очень понял, я же как раз его и использую

File: packages/plasma-hope/src/components/Checkbox/Checkbox.tsx (hunk 3 above)
Comment (kayman233) [reply packages/plasma-hope/src/components/Checkbox/Checkbox.tsx:144]: отписался выше, зачем тут это использую

File: packages/plasma-core/src/components/Checkbox/Checkbox.component-test.tsx (hunk 2 above)
Comment (kayman233) [reply packages/plasma-core/src/components/Checkbox/Checkbox.component-test.tsx:111]: у нас же тут как раз и проверяется поведение, когда мы хотим, чтобы одна строка была. А в остальных тестах по суть и так уже проверяется то, что многострочность работает(поэтому как раз и оставил по дефолту многострочность)

File: packages/plasma-core/src/components/Basebox/Basebox.tsx (hunk 1 above)
Comment (Yeti-or) [reply packages/plasma-core/src/components/Basebox/Basebox.tsx:68]: это меня проглучило

File: cypress/snapshots/ui/components/Checkbox/CheckboxBase.component-test.tsx/plasma-core Checkbox -- squeezes.snap.png
Comment (Yeti-or) [reply cypress/snapshots/ui/components/Checkbox/CheckboxBase.component-test.tsx/plasma-core Checkbox -- squeezes.snap.png:1]: Давай напишием в описании что произошло два изменения:
1) поправили label / description - было / стало
2) добавили свой-во: без него / с ним

**Pull request merged**